accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/rest/BasicRestControllerAdvice.java

```diff
@@ -94,11 +98,15 @@
     @ExceptionHandler(value = AccessionMergedException.class)
     public ResponseEntity<ErrorMessage> handleMergeExceptions(AccessionMergedException ex) {
         logger.error(ex.getMessage(), ex);
-        String originalRequestUrl = ServletUriComponentsBuilder.fromCurrentRequestUri().toUriString();
-        return ResponseEntity.status(HttpStatus.MOVED_PERMANENTLY)
-                .location(URI.create(originalRequestUrl.replace(ex.getOriginAccessionId(), ex
-                        .getDestinationAccessionId()))).body(new ErrorMessage(HttpStatus.MOVED_PERMANENTLY, ex,
-                        ex.getMessage()));
+        String originalRequestUrl = httpServletRequest.getRequestURL().toString();
```

Comment:
This local variable is only used inside the if() block. It should be moved in there.

Reply:
fixed on db42c25